src/app/qgsattributetabledialog.cpp

```diff
@@ -351,6 +353,26 @@ void QgsAttributeTableDialog::updateFieldFromExpressionSelected()
   runFieldCalculation( mLayer, mFieldCombo->currentText(), mUpdateExpressionText->currentField(), filteredIds );
 }
 
+void QgsAttributeTableDialog::viewWillShowContextMenu( QMenu* menu, QModelIndex atIndex )
+{
+  if ( menu )
+  {
+    menu->addAction( tr( "Zoom to feature" ), this, SLOT( zoomToFeature() ) );
+  }
```

Comment:
Could the _QgsFeatureId_ be determined here solely using `QModelIndex atIndex`? It could then be passed to `zoomToFeature()` via a _QSignalMapper_, _QAction.setData()_ or some other means.

This would avoid exposing _filterModel()_ and _tableView()_. But I don't even know if that is an issue at all.